src/utl/src/ScopedTemporaryFile.cpp

```diff
@@ -33,4 +36,61 @@ ScopedTemporaryFile::~ScopedTemporaryFile()
   }
 }
 
+StreamHandler::StreamHandler(const char* filename,
+                             std::ios_base::iostate flag,
+                             std::ios_base::openmode mode)
+    : filename_(filename), flag_(flag), mode_(mode)
+{
+  if (fs::exists(filename_)) {
+    fs::remove(filename_);
+  }
+  std::string tmp_filename = filename_ + ".tmp";
+  if (fs::exists(tmp_filename)) {
+    fs::remove(tmp_filename);
+  }
+  os_.exceptions(flag | std::ofstream::failbit);
+  os_.open(tmp_filename, mode);
+}
+
+StreamHandler::~StreamHandler()
+{
+  if (os_.is_open()) {
+    os_.close();
+  }
+  std::string tmp_filename = filename_ + ".tmp";
+  fs::rename(tmp_filename, filename_);
+}
+
+std::ofstream& StreamHandler::getStream()
+{
+  return os_;
+}
+
+FileHandler::FileHandler(const char* filename, const char* mode)
+    : filename_(filename), mode_(mode)
+{
+  if (fs::exists(filename_)) {
+    fs::remove(filename_);
```

Comment:
Remove the mode argument.

It implies that "r" and "rb" and "w" and "wb" are all supported, but they are not. Bad unexpected things will happen...

Reply:
I'll change mode to binary, which will be boolean type and if its set then file will be opened in `wb` mode otherwise in `w`